cpp/mrc/include/mrc/node/rx_node.hpp

```diff
@@ -235,6 +235,7 @@ class RxNodeComponent : public WritableProvider<InputT>, public WritableAcceptor
                 this->get_writable_edge()->await_write(std::move(message));
             },
             [this](std::exception_ptr ptr) {
+                WritableAcceptor<OutputT>::release_edge_connection();
```

Comment:
Can you add a test to `test_edges.cpp` that checks for this? Should be pretty easy to throw an error in a node component. You will get an error if its not released correctly.

Reply:
I can, but `test_segment.cpp` or `test_node.cpp` might be better locations for this. The reason is that in order to have a context set it needs to run inside of a segment, and there are currently no tests like this in `test_edges.cpp`

Reply:
lets put it in `test_node.cpp`